lib/date_time/date_time.c

```diff
@@ -268,69 +284,69 @@ static void new_date_time_get(void)
 
 		err = current_time_check();
 		if (err == 0) {
-			LOG_DBG("Time successfully obtained");
+			LOG_DBG("Using previously obtained time");
 			initial_valid_time = true;
+			date_time_schedule_update();
 			date_time_notify_event(&evt);
 			continue;
 		}
 
 		LOG_DBG("Current time not valid");
 
-#if defined(CONFIG_DATE_TIME_NTP)
-		LOG_DBG("Fallback on NTP server");
+#if defined(CONFIG_DATE_TIME_MODEM)
+		LOG_DBG("Getting time from cellular network");
 
-		err = time_NTP_server_get();
+		err = time_modem_get();
 		if (err == 0) {
-			LOG_DBG("Time from NTP server obtained");
+			LOG_DBG("Time from cellular network obtained");
 			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_NTP;
+			evt.type = DATE_TIME_OBTAINED_MODEM;
+			date_time_schedule_update();
 			date_time_notify_event(&evt);
 			continue;
 		}
 
-		LOG_DBG("Not getting time from NTP server");
+		LOG_DBG("Did not get time from cellular network. "
+			"This is normal as some cellular networks don't provide it.");
 #endif
-#if defined(CONFIG_DATE_TIME_MODEM)
-		LOG_DBG("Fallback on cellular network time");
+#if defined(CONFIG_DATE_TIME_NTP)
+		LOG_DBG("Getting time from NTP server");
 
-		err = time_modem_get();
+		err = time_NTP_server_get();
 		if (err == 0) {
-			LOG_DBG("Time from cellular network obtained");
+			LOG_DBG("Time from NTP server obtained");
 			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_MODEM;
+			evt.type = DATE_TIME_OBTAINED_NTP;
+			date_time_schedule_update();
 			date_time_notify_event(&evt);
 			continue;
 		}
 
-		LOG_DBG("Not getting cellular network time");
+		LOG_DBG("Did not get time from NTP server");
 #endif
-		LOG_DBG("Not getting time from any time source");
+		LOG_DBG("Did not get time from any time source");
 
 		evt.type = DATE_TIME_NOT_OBTAINED;
+		date_time_schedule_update();
 		date_time_notify_event(&evt);
 	}
 }
 
-K_THREAD_DEFINE(time_thread, CONFIG_DATE_TIME_THREAD_SIZE,
+K_THREAD_DEFINE(time_thread, CONFIG_DATE_TIME_THREAD_STACK_SIZE,
```

Comment:
I rebased nrf master to this PR branch and started to see time_thread crashes (saw them with some of my other branches earlier first). I noticed that @lemrey had taken new AT interface into use which likely increased the stack requirement:
https://github.com/nrfconnect/sdk-nrf/commit/6513b764ffe534eeb31d9f670d91f33706a7e840

I've noticed that @jtguggedal has a commit in [PR#5741](https://github.com/nrfconnect/sdk-nrf/pull/5741) to increase the stack size a bit:
https://github.com/nrfconnect/sdk-nrf/pull/5741/commits/e1b72f61c212fb4fe8d7a25fa08d80c71429e614
I've cherry-picked this commit into this PR because I expect this PR to be completed before PR#5741 and I have few use cases which need this stack change (I know I can increase the configured value but this is more convenient).

Does this make sense?

Reply:
Hi, yes I have seen the same; my bad we need to bump that stack size a bit.